[needs-docs] Reorganisation of contextual menu for group and layer

slarosa opened the pull request:
## Description
Another change to reorganize the contextual menu for group and layers. Please let me know if it looks good to you.

<img width="310" alt="schermata 2018-03-22 alle 11 31 40" src="https://user-images.githubusercontent.com/1374682/37765466-79620f98-2dc4-11e8-8a16-f181999f365c.png">

<img width="249" alt="schermata 2018-03-22 alle 11 31 32" src="https://user-images.githubusercontent.com/1374682/37765475-7e597054-2dc4-11e8-8694-434f45a860a1.png">

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

slarosa committed:
Reorganisation of contextual menu for group and layer

saberraz commented:
@slarosa there was a discussion a while back to remove Overview, now that we have multiple map canvases and also map themes....maybe a good time to remove it while you are clearing up the menu?

nirvn commented:
Please do not remove overview, it's a basic tool many uses on a daily basis, map views aren't a replacement to this UX.

I'll comment on item positioning in an hour or so. I like the idea of shuffling things around.

BTW I noticed not all group menu items are in camel case, that's worth fixing.

nirvn commented:
@slarosa , I'd suggest the following:

```
_________________________

Zoom to Group
Copy Group
Rename
---
Add Group
Remove
---
Set Group CRS
---
Mutually Exclusive Group
Check Group & all its Children
Uncheck Group & all its Children
---
Export >
_________________________


_________________________

Zoom to Layer
Show in Overview
Show Feature Count
Copy Layer
Rename
---
Duplicate
Remove
---
Open Attribute Table
Filter...
---
Styles >
Properties
---
Set Layer Scale Visibility
Set Layer CRS
Set Project CRS from Layer
---
Export >
_________________________

```

I especially think the add group / remove and duplicate (layer) / remove are a natural pairs.

nirvn commented:
@slarosa , I also think the group's "save as layer definition..." menu item should be put behind an export sub-menu. That'll harmonize it with layer, and while we only have one export option ATM, who knows, we might have more in the future.

slarosa commented:
@nirvn both proposals (my and your) look good to me as the actual order of contextual menu is very messy. I had some doubt on ```Remove``` and ```Properties``` entries if them fit better to the bottom, but I think your suggestion work fine. 

Regarding Overview dock, I am in favor to remove it, for the same reason that @saberraz has said, but we eventually can do this in a separate PR.

nirvn commented:
Yep, reviewing the overview dock is IMHO something that would have to go through some sort of a voting process (probably a QEP to propose an alternative UX or something).

slarosa commented:
@nirvn please could you give a look at the new changes, thanks.

<img width="307" alt="schermata 2018-03-23 alle 19 28 11" src="https://user-images.githubusercontent.com/1374682/37847538-6968dbae-2ed1-11e8-82a0-b54b9f45b559.png">
<img width="243" alt="schermata 2018-03-23 alle 19 27 59" src="https://user-images.githubusercontent.com/1374682/37847541-6e367042-2ed1-11e8-8ac6-7d21d56cf66f.png">

nyalldawson commented:
Nice work!

There's some more inconsistencies in these menus which should also be addressed:

- bad capitalization, e.g. "data" in "Set WMS data", the "check all..." items
- inconsistent phrasing. Some items mention "group" or "layer", others don't. So we have "copy group" and then "rename". Either they should all use the noun or none of them.
- missing ellipsis, e.g. in "set group crs", "properties", "remove"

MieWinstrup commented:
I really like the idea of reordering the context menu! 

My comments:

- 'Move to Top level' could be named 'Ungroup' or 'Move out of group'
- I'm wondering if there is a reason why two of the items ends with '...' ('Filter...' and 'Edit Virtuel Layer...'). If there isn't a reason I think the '...' should be removed.
- There is a lot of important functionality in 'Properties' - I wonder if it would be an idea to place the 'group' with Styles and Properties just above the 'Export'?
- If there were icons for all of the functions, that would be really nice - I would be very happy to come up with some suggestions for icons.

nirvn commented:
@slarosa , great responsiveness, thanks. I'm good with when the capitalization issue is fixed. 

Regarding inconsistent use of ellipsis, I'd be in favor of adding the missing ones instead of removing the existing ones.

Regarding inconsistent phrasing, +1 to address this, can be done in a separate PR or here.

slarosa commented:
@MieWinstrup wrote:
> There is a lot of important functionality in 'Properties' - I wonder if it would be an idea to place the 'group' with Styles and Properties just above the 'Export'?

@nirvn What you think about this changes? Even I agree that should be a good move.

nirvn commented:
@andreasneumann , -1 to remove "Set layer CRS", it's something I'm using _all_ _the_ _time_ :smile:

andreasneumann commented:
@nirvn Why? if your data contains proper CRS information (SRS definition in PostgreSQL or .prj files for shapes) - you won't ever need that setting. QGIS will then auto-detect that information.

nyalldawson commented:
Also -1 to remove the crs option, because it works for multi layer selections. Try setting the crs for 20 layers otherwise!

nirvn commented:
@andreasneumann , _"If your data contains proper CRS information..."_ ... if only :wink: in my part of the world, where we've had several CRS over time to cover the same region, and users not being very accurate or careful, this shortcut is a huge time saver.

andreasneumann commented:
@nyalldawson - I agree for the group it makes sense, but not on individual layers. Also the group context menu is smaller, but the layer context menu is already too big for my taste.

andreasneumann commented:
Ok - we probably don't reach consensus here - but please don't add more entries to the context menus ;-) unless there are really good reasons and that functionality is really used frequently.

nirvn commented:
@andreasneumann , we could move these CRS items into a "Projection >" sub-menu, that said the idea would probably deserve its own PR and chat :)

nyalldawson commented:
Regarding properties: I believe for ideal ux this should be the final item in the menu (fitts law)

slarosa commented:
I done some more tweak addressing some comment.

<img width="440" alt="schermata 2018-03-26 alle 17 30 59" src="https://user-images.githubusercontent.com/1374682/37916358-e353711c-311b-11e8-9d35-9cb9a8cd9c3a.png">
<img width="233" alt="schermata 2018-03-26 alle 17 34 19" src="https://user-images.githubusercontent.com/1374682/37916363-e8157632-311b-11e8-8715-146539553d75.png">

MieWinstrup commented:
I like it! 
How about moving the new 'Copy Layer' down to 'Duplicate' and 'Remove'? 
Is there a reason for the '...' after 'Remove'?

nyalldawson commented:
@MieWinstrup

Standard UI HIG is that the "ellipsis" (...) should be used whenever an action requires further confirmation or settings before it's performed. So in this case Remove needs it, because there's a prompt before the layer removal actually happens.

slarosa committed:
capitalization, ellipsis and minor fixes

nirvn commented:
Restarted Travis. All OK to merge when green?

nirvn commented:
@slarosa , merging this now. IMHO, we still need to address the noun issue raised by @nyalldawson , another PR would be most welcome.
